test/commonkafka.py

```diff
@@ -27,29 +31,68 @@
 handler.setFormatter(formatter)
 logger.addHandler(handler)
 
+
 def create_kafka_connector(setup, params):
     '''
     Create kafka connect connector using kafka connect REST API
     '''
-    response = requests.post(url=setup["kafka_connect_url"]+"/connectors", data=json.dumps(params),
-                      headers={'Accept': 'application/json', 'Content-Type': 'application/json'})
+    response = requests.post(url=setup["kafka_connect_url"] + "/connectors", data=json.dumps(params),
+                             headers={'Accept': 'application/json', 'Content-Type': 'application/json'})
 
     if response.status_code == 201:
         logger.info("Created connector successfully - " + json.dumps(params))
         return True
 
     return False
 
-def delete_kafka_connector(setup, params):
+
+def delete_kafka_connector(setup, connector):
     '''
     Delete kafka connect connector using kafka connect REST API
     '''
-    response = requests.delete(url=setup["kafka_connect_url"]+"/connectors/" + params["name"],
-                        headers={'Accept': 'application/json', 'Content-Type': 'application/json'})
+    if not isinstance(connector, str):
+        connector = connector['name']
+    response = requests.delete(url=setup["kafka_connect_url"] + "/connectors/" + connector,
+                               headers={'Accept': 'application/json', 'Content-Type': 'application/json'})
     if response.status_code == 204:
-        logger.info("Deleted connector successfully - " + json.dumps(params))
+        logger.info("Deleted connector successfully - " + json.dumps(connector))
         return True
     else:
-        logger.error("failed to create connector", param)
+        logger.error("failed to delete connector", connector)
         logger.error(response.status_code)
-        return False
\ No newline at end of file
+        return False
+
+
+def pytest_configure(setup):
+    # Generate data
+    now = datetime.now()
+    msg = {'timestamp': datetime.timestamp(now)}
+    generate_data(setup, msg)
+    setup['timestamp'] = msg["timestamp"]
+
+    # Launch all connectors for tests
+    for param in connect_params:
+        create_kafka_connector(setup, param)
+    # wait for data to be ingested to Splunk
+    time.sleep(80)
+
+
+def generate_data(setup, msg):
+    producer = KafkaProducer(bootstrap_servers=setup["kafka_broker_url"],
+                             value_serializer=lambda v: json.dumps(v).encode('utf-8'))
+    producer.send(setup["kafka_topic"], msg)
+    producer.flush()
+
+
+def pytest_unconfigure(setup):
+    # Delete launched connectors
+    content = requests.get(url=setup["kafka_connect_url"] + "/connectors",
```

Comment:
Deleting all deployed connectors are not always what we want to do. we want to delete connectors that were started by this test.

Reply:
The problem is in test_crud.py for example, if the test failed on line 82, the delete_kafka_connector will never be executed, the connector will stay there and affect the test result for other test cases.

Reply:
how are these lingering connectors affecting the other test? it should be made not to.

Reply:
we can solve these lingering connectors by adding delete calls to pytest_unconfigure function in conftest.py.

Reply:
we know exactly the names of connectors we are deploying during test

Reply:
Has this issue been resolved? It looks like we have an unconfigure method and setup_class method in TestCrud class. We have two implementations for the same problem of deleting lingering connectors. Ideally, we would have one unconfigure/destroy method the deletes the connectors for that particular test class.

Reply:
I solved this issue with a different approach. I am collecting all the running connectors here using this method, and in TestCrud class, whenever we create a new connector I add it in the memory and after test class finish, check any of the new created connector still running, if yes, i delete them. In this case, there won't be any zombie connectors left behind.